# Patch to adjust CFLAGS, link command line tool to shared library and do not install static
# Copyright (C) 2016 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at http://sotirov-bg.net/slackpack/
#
# $Id: kvazaar-0.6.1.diff,v 1.1 2016/09/22 10:21:44 gsotirov Exp $
#
diff -urNad kvazaar-0.6.1-orig/src/Makefile kvazaar-0.6.1/src/Makefile
--- kvazaar-0.6.1-orig/src/Makefile	2015-09-16 18:42:20.000000000 +0300
+++ kvazaar-0.6.1/src/Makefile	2016-09-22 13:18:04.000000000 +0300
@@ -50,10 +50,10 @@
 # clang-4.3.0: slight benefit from -O3, more from -flto
 CFLAGS += -O2 -g -ftree-vectorize
 
-CFLAGS += -fvisibility=hidden -std=gnu99
+CFLAGS += -std=gnu99
 CFLAGS += -DKVZ_DLL_EXPORTS
 CFLAGS += $(INCLUDEDIRS) $(WARNINGS)
-LDFLAGS += -O2 -fvisibility=hidden -lm -pthread
+LDFLAGS += -O2 -lm -pthread
 
 ARCH ?= $(shell uname -m)
 
@@ -116,7 +116,7 @@
     ASFLAGS += -f elf32
   endif
   LIBKVAZAAR_SHARED = $(LIB)
-  INSTALL_TARGETS += install-lib install-static
+  INSTALL_TARGETS += install-lib
 endif
 
 # Flags shared across systems
@@ -255,8 +255,8 @@
                      -compatibility_version $(VER_MAJOR) \
                      -install_name $(LIBDIR)/$@
 
-$(PROG): $(MAIN_OBJS) $(STATIC)
-	$(LD) $^ $(LDFLAGS) $(LIBS) -o $@
+$(PROG): $(MAIN_OBJS) $(LIBKVAZAAR_SHARED)
+	$(LD) $^ $(LDFLAGS) $(LIBS) -L. -lkvazaar -o $@
 
 $(STATIC): $(OBJS)
 	$(RM) $@
@@ -265,6 +265,8 @@
 
 $(LIB) $(DLL) $(DYLIB): $(SHARED_OBJS)
 	$(LD) $^ $(LDFLAGS) $(LIBS) -o $@
+	$(LN_S) -f -T $(LIB) $(SO).$(VER_MAJOR)
+	$(LN_S) -f -T $(LIB) $(SO)
 
 $(DEBUG): $(DEBUG_OBJS)
 	$(LD) $^ $(LDFLAGS) $(LIBS) -o $@
@@ -356,6 +358,8 @@
 clean:
 	$(RM) $(RELEASE_OBJS) $(SHARED_OBJS) $(DEBUG_OBJS) $(TESTS_OBJS) $(DEPS)
 	$(RM) $(PROG) $(DEBUG) $(TESTS) $(STATIC) $(LIB) $(DLL) $(IMPLIB) $(DYLIB)
+	$(RM) -f $(SO).$(VER_MAJOR) $(SO)
+
 
 ifneq ($(MAKECMDGOALS),clean)
 -include $(DEPS)
